### ISSUE TITLE
New release on PyPI

### ISSUE THREAD

**Issue by absporl** [pyca/pynacl#839]
Are there plans to release the current version of `main` anytime soon? The current release 1.5.0 on PyPI is from January 2022. I've run into an issue that seems fixed by the current commit, probably due to the updated bundled libsodium, so it would be helpful to have an updated version on PyPI as well.

**Comment by absporl** [comment]
That'd be very helpful, thanks! Do you have an idea when that could be? I'm working on a package that needs the current pynacl master, but I'd like to avoid pointing the dependency spec to the git master.

**Comment by santiagobasulto** [comment]
@reaperhulk seems like the latest version (1.5.0) has been released a few weeks ago, is it safe to close this issue?

**Comment by reaperhulk** [comment]
Yeah no I sitll need to get to this, we should definitely leave it open 😄

**Comment by santiagobasulto** [comment]
@reaperhulk honest question, what's the difference between PyNaCL and [pysodium](https://github.com/stef/pysodium)? I've been using Pysodium lately, seems fairly well maintained.

**Comment by gonatienza** [comment]
@santiagobasulto , they are both listed under libsodium [bindings](https://libsodium.gitbook.io/doc/bindings_for_other_languages).

The main three differences for me:
1. pynacl and cryptography are both from [pyca](https://github.com/pyca), which is owned by a group of contributors and not an individual contributor.
2. pysodium does not include the libsodium release for which the bindings were written.  I.e, among other possible errors, if you pip install it but libsodium is not there you will get a "Unable to find libsodium" error.
3. pynacl includes some high level wrappers and documentation that make high level implementations safe and not bound to common errors implementation errors (like a basic nonce reuse).

HTH!

**Comment by santiagobasulto** [comment]
Thanks @gonatienza !

**Comment by bibi21000** [comment]
@reaperhulk ... i agree ... we need it :)

I've implemented a python file like interface to encrypt with PyNacl (and others). More details here : https://github.com/bibi21000/NaclFile.

I've done some benchmarks to compare with other cryptographics tools  : https://github.com/bibi21000/CofferFile/blob/main/BENCHMARK.md 

PyNacl is the fastest and produce the smallest files

Congrats and ty 👍

**Comment by finnagin** [comment]
+1

Would love to see a new release to make the windows arm64 wheels available.

**Comment by cdce8p** [comment]
A new release would be great. Testing with 3.14 is currently blocked for us since https://github.com/pyca/pynacl/pull/848 isn't released yet. We're already at `b2`, only a few months until the release candidates. The earlier this is resolved here, the earlier the testing can continue in downstream projects.

**Comment by khmyznikov** [comment]
@cdce8p can you break down what we can help with to make this happen? [This one](https://github.com/python/cpython/issues/118803)?

**Comment by cdce8p** [comment]
> [@cdce8p](https://github.com/cdce8p) can you break down what we can help with to make this happen? [This one](https://github.com/python/cpython/issues/118803)?

The fix is already part of main, so all that's needed here AFAICT at the moment would be to publish a new release to PyPI. https://github.com/pyca/pynacl/commit/d33028e43b814615a33e231925eaddb0f679fa2b

**Comment by khmyznikov** [comment]
Any news about the new PyPi release?

**Comment by reaperhulk** [comment]
I’ll be doing one right after cffi 2.0 is released.

**Comment by h-vetinari** [comment]
> I’ll be doing one right after cffi 2.0 is released.

cffi 2.0 might still take [a while](https://github.com/python-cffi/cffi/issues/187) yet, perhaps into October or however long it takes. Assuming pynacl releases are not too arduous, a humble suggestion would be: release 1.6.0 without free-threading support (but all the other accumulated fixes) ASAP, and then do 1.6.1 once cffi 2.0 is out.

**Comment by ngoldbaum** [comment]
3.14 and 3.14t support are merged. If there are any tasks that need doing to enable a release, I'd be happy to help out.

**Comment by reaperhulk** [comment]
1.6.0 is released, huge thanks to all who contributed, especially @ngoldbaum for his extensive work on free-threaded Python.
